src/connector/src/parser/additional_columns.rs

```diff
@@ -99,14 +167,17 @@ pub static CDC_BACKFILL_TABLE_ADDITIONAL_COLUMNS: LazyLock<Option<HashSet<&'stat
         ]))
     });
 
-pub fn get_supported_additional_columns(
-    connector_name: &str,
+pub fn get_supported_additional_columns<'a>(
+    connector_name: &'a str,
+    format: &FormatType,
     is_cdc_backfill: bool,
-) -> Option<&HashSet<&'static str>> {
+) -> Option<&'a HashSet<&'static str>> {
     if is_cdc_backfill {
         CDC_BACKFILL_TABLE_ADDITIONAL_COLUMNS.as_ref()
     } else {
-        COMPATIBLE_ADDITIONAL_COLUMNS.get(connector_name)
+        COMPATIBLE_ADDITIONAL_COLUMNS
+            .get(connector_name)
+            .and_then(|map| map.get(format))
```

Comment:
I think add `format` to the hashmap makes it harder to manage.

```suggestion
        let compactible_additional_columns = COMPATIBLE_ADDITIONAL_COLUMNS
            .get(connector_name)?;
        match format { // check format ... }
```
may be better

Reply:
Because "connector metadata" and "format metadata" are relatively orthogonal.

Reply:
make sense.
But I am not sure whether we allow including `key` for debezium format. This is the only concern I use the nested structure.